src/senaite/core/browser/attachment/attachment.py

```diff
@@ -441,22 +422,11 @@ def get_analyses(self):
         """Returns a list of analyses from the AR
         """
         analyses = self.context.getAnalyses(full_objects=True)
-        return filter(self.is_analysis_attachment_allowed, analyses)
-
-    def is_analysis_attachment_allowed(self, analysis):
-        """Checks if the analysis
-        """
-        if analysis.getAttachmentOption() not in ["p", "r"]:
-            return False
-        if api.get_workflow_status_of(analysis) in ["retracted"]:
-            return False
-        return True
+        return filter(api.is_active, analyses)
```

Comment:
Note that `api.is_active` only checks for review statuses `cancelled` and `invalid`. Analyses have other status that can be considered as not suitable for the addition of attachments: "retracted", "rejected", "verified", "published"
Therefore, I think a better option would be:
```python
analyses = filter(api.is_active, analyses)
analyses = filter(lambda a: not IVerified.providedBy(a), analyses)
```
Another option (maybe better) would be to add the following function on `api.analysis`:

```python
def is_processed(analysis):
    if not api.is_active(analysis):
        return True
    return IVerified.providedBy(analysis)

def is_ongoing(analysis):
    return not is_processed(analysis)
```

So this line would end up being:
```python
return filter(analysis_api.is_ongoing, analyses)
```

Another option would be generalize `api.is_active` by checking first for `cancelled` and `invalid` and if the object status is none of those two, then look for an adapter for the object that tells the statuses that need to be considered as not active. With this approach, we could always use `api.is_active` regardless of the workflow the object is currently bound to. Also, it would allow us to modify the status considered as "inactive" in other add-ons.

Reply:
This is addressed on https://github.com/senaite/senaite.core/pull/1710